src/WebJobs.Script/Workers/Http/HttpWorkerProcess.cs

```diff
@@ -54,11 +54,8 @@ internal override Process CreateWorkerProcess()
         internal override void HandleWorkerProcessExitError(WorkerProcessExitException langExc)
         {
             // The subscriber of WorkerErrorEvent is expected to Dispose() the errored channel
-            if (langExc != null && langExc.ExitCode != -1)
-            {
-                _workerProcessLogger.LogDebug(langExc, $"Language Worker Process exited.", _workerProcessArguments.ExecutablePath);
-                _eventManager.Publish(new HttpWorkerErrorEvent(_workerId, langExc));
-            }
+            _workerProcessLogger.LogDebug(langExc, $"Language Worker Process exited. Pid={langExc.Pid}.", _workerProcessArguments.ExecutablePath);
```

Comment:
Are we always sure that we have langExc  value?

Reply:
The fact that we had a check before leads me to believe we wanted to handle the possibility that this was `null`. If that isn't the case, we should have a proper argument validation here.

Reply:
This method is only called here, where an exception is explicitly created: https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/Workers/ProcessManagement/WorkerProcess.cs#L139

Adding argument validation!

Reply:
Yeah, if the expectation is that this wouldn't be called without a `null` argument, let's have this method validate that

Reply:
@fabiocav and @amamounelsayed - thoughts now? also if you have a better suggestion for the ArgumentNullException input very much appreciated!

Reply:
The expectation is that you'd pass the argument name into that constructor.

Change to :
``` csharp
throw new ArgumentNullException(nameof(langExc));
```

Though, a better argument name would be nice :)

Reply:
thanks!! do you mind reviewing again?